frame/dapps-staking/src/pallet/mod.rs

```diff
@@ -280,6 +296,16 @@ pub mod pallet {
         AlreadyUsedDeveloperAccount,
         /// Unexpected state error, used to abort transaction
         UnexpectedState,
+        /// Report issue on github if this is ever emitted
+        UnknownStartStakingData,
+        /// Report issue on github if this is ever emitted
+        UnknownEraReward,
```

Comment:
I think we should keep these 'never should happen errors' minimized.

Not sure if that is the best approach, but I used 'UnexpectedState' with an additional message print to handle such situations.

Having many errors that 'should never occur' seems like adding too much additional complication.

Please share your opinion.

Reply:
I agree and we should minimise this kind of error. However I propose to keep it until it is ready for production. Good to have it for debugging at this stage